packages/detector.go

```diff
@@ -11,67 +11,143 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+
+// Package packages implements package manager detection and package
+// manager specific output formatting.
 package packages
 
 import (
+	"bytes"
 	"errors"
 	"fmt"
-	"os/exec"
-	"syscall"
+	"os"
+	"regexp"
+	"strconv"
 
 	"github.com/sirupsen/logrus"
 )
 
-var execCommand = exec.Command
+// SupportedPackageManagers represents the standard error string used
+// when OS package maanger can not be identified.
+const SupportedPackageManagers = "No supported package manager found; apk, dpkg, dnf or yum installed?"
```

Comment:
A question here feels a little off I think. Maybe something like this?? 
```suggestion
const SupportedPackageManagers = "No supported package manager could be auto detected. Supported versions are apk, dpkg, dnf or yum."
```

Reply:
Fixed.